build/build.go

```diff
@@ -0,0 +1,12 @@
+/*
+Package build contains build specific information.
+*/
+package build
+
+// gitRevision is injected at build time.
+var gitRevision string
+
+// GetGitRevision retrieves the revision injected into the build at build time.
+func GetGitRevision() string {
```

Comment:
Can you add a todo comment to use `runtime/debug.ReadBuildInfo` when we no longer support go1.17 please?

Reply:
In go1.18 BuildInfo will return vcs for example:

```go
package main

import (
	"fmt"
	"runtime/debug"
)

func main() {
	var gitCommitID string
	var gitCommitTS string
	info, _ := debug.ReadBuildInfo()
	for _, setting := range info.Settings {
		fmt.Printf("setting: name=%s value=%s\n", setting.Key, setting.Value)
		switch setting.Key {
		case "vcs.revision":
			gitCommitID = setting.Value
		case "vcs.time":
			gitCommitTS = setting.Value
		}
	}
	fmt.Println("git commit:", gitCommitID)
	fmt.Println("git commit time:", gitCommitTS)
}
```

```ShellSession
[16:10:42]-[~/go/src/github.com/tinkerbell/boots]─[manny@zennix]> go build -o t; ./t
setting: name=-compiler value=gc
setting: name=CGO_ENABLED value=1
setting: name=CGO_CFLAGS value=
setting: name=CGO_CPPFLAGS value=
setting: name=CGO_CXXFLAGS value=
setting: name=CGO_LDFLAGS value=
setting: name=GOARCH value=amd64
setting: name=GOOS value=linux
setting: name=GOAMD64 value=v1
setting: name=vcs value=git
setting: name=vcs.revision value=08af73b8675e50c27e3422daff5835fab3ddbb90
setting: name=vcs.time value=2022-04-27T20:04:23Z
setting: name=vcs.modified value=true
git commit: 08af73b8675e50c27e3422daff5835fab3ddbb90
git commit time: 2022-04-27T20:04:23Z
```

So when we switch to 1.18+ only we can do away with passing vcs info via compiler/linker flags and just pull it out from debug.

Reply:
Thats nifty. I hadn't seen that in the release notes, thanks.

Reply:
Resolved.